src/test/java/com/googleinterns/zoomtube/servlets/TranscriptServletTest.java

```diff
@@ -0,0 +1,264 @@
+    assertThat(actualJsonArray).isEqualTo(expectedArrayList);
+  }
+
+  @Test
+  public void doGet_returnsLectureForLongVideoFromDatastore() throws ServletException, IOException {
+    putTranscriptLinesInDatastore(longVideoTranscriptLines, LECTURE_ID_A);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_A);
+
+    servlet.doGet(request, response);
+
+    String actualJson = lectureTranscript.toString();
+    List<TranscriptLine> expectedArrayList = longVideoTranscriptLines;
+    List<TranscriptLine> actualJsonArray = transcriptLines(actualJson);
+    assertThat(actualJsonArray.size()).isEqualTo(expectedArrayList.size());
+  }
+
+  @Test
+  public void doPost_persistDataInDatastoreForLongVideo() throws ServletException, IOException {
+    when(request.getParameter(TranscriptServlet.PARAM_VIDEO_ID)).thenReturn(LONG_VIDEO_ID);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_C);
+
+    servlet.doPost(request, response);
+
+    int actualQueryCount = entitiesInDatastoreCount(LECTURE_ID_C);
+    int expectedQueryCount = (transcriptLines(LONG_VIDEO_JSON)).size();
+    assertThat(actualQueryCount).isEqualTo(expectedQueryCount);
+  }
+
+  @Test
+  public void doGet_onlyOtherLecturesInDatastore_GetNoLectures()
+      throws ServletException, IOException {
+    putTranscriptLinesInDatastore(shortVideoTranscriptLines, LECTURE_ID_B);
+    putTranscriptLinesInDatastore(longVideoTranscriptLines, LECTURE_ID_A);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_C);
+
+    servlet.doGet(request, response);
+
+    String actualJson = lectureTranscript.toString();
+    List<TranscriptLine> actualJsonArray = transcriptLines(actualJson);
+    assertThat(actualJsonArray.size()).isEqualTo(0);
+  }
+
+  @Test
+  public void doGet_twoLecturesInDatastore_returnsOneLecture()
+      throws ServletException, IOException {
+    putTranscriptLinesInDatastore(shortVideoTranscriptLines, LECTURE_ID_B);
+    putTranscriptLinesInDatastore(longVideoTranscriptLines, LECTURE_ID_A);
+    when(request.getParameter(TranscriptServlet.PARAM_LECTURE_ID)).thenReturn(LECTURE_ID_A);
+
+    servlet.doGet(request, response);
+
+    String actualJson = lectureTranscript.toString();
+    List<TranscriptLine> expectedArrayList = longVideoTranscriptLines;
+    List<TranscriptLine> actualJsonArray = transcriptLines(actualJson);
+    assertThat(actualJsonArray.size()).isEqualTo(expectedArrayList.size());
+  }
+
+  private static List<TranscriptLine> transcriptLines(String transcriptLinesJson) {
+    Gson gson = new GsonBuilder().registerTypeAdapterFactory(GenerateTypeAdapter.FACTORY).create();
+    return (ArrayList<TranscriptLine>) gson.fromJson(
+      transcriptLinesJson, (new ArrayList<List<TranscriptLine>>().getClass()));
+  }
+
+
+  private void putTranscriptLinesInDatastore(List<TranscriptLine> transcriptLineArray, String lectureKeyId) {
```

Comment:
Please remember that "array" is a filler word. It is not necessary. Typically if you want to know the object type you look at the declaration. For example, for lectureKeyId you don't write lectureKeyIdString. So this is similar.

Please look at: https://docs.google.com/document/d/1_ldClEZLsI1cMpQl_BfeIlkjTsE2IiJ2PgvhABfO6vs/edit

Reply:
fixed